tic_tac_toe/game.py

```diff
@@ -1,104 +1,145 @@
-# internal helpers
-def _position_is_empty_in_board(position, board):
-    """
-    Checks if given position is empty ("-") in the board.
-
-    :param position: Two-elements tuple representing a
-                     position in the board. Example: (0, 1)
-    :param board: Game board.
+from tic_tac_toe.exceptions import InvalidMovement, GameOver
 
-    Returns True if given position is empty, False otherwise.
-    """
-    pass
 
 
-def _position_is_valid(position):
-    """
-    Checks if given position is a valid. To consider a position as valid, it
-    must be a two-elements tuple, containing values from 0 to 2.
-    Examples of valid positions: (0,0), (1,0)
-    Examples of invalid positions: (0,0,1), (9,8), False
+def _position_is_empty_in_board(position, board):
+    pos1 = position[0]
+    pos2 = position[1]
+    if board[pos1][pos2] == "-":
+        return True
+    else:
+        return False
 
-    :param position: Two-elements tuple representing a
-                     position in the board. Example: (0, 1)
 
-    Returns True if given position is valid, False otherwise.
-    """
-    pass
 
+def _position_is_valid(position):
+    valid_positions = [(row, col) for row in range(3) for col in range(3)]
+    
+    if position in valid_positions:
+        return True
+    else:
+        return False
+    
+    
 
 def _board_is_full(board):
-    """
-    Returns True if all positions in given board are occupied.
+    for row in board:
+        for pos in row:
+            if pos == '-':
+                return False
+    else:
+        return True
 
-    :param board: Game board.
-    """
-    pass
 
 
 def _is_winning_combination(board, combination, player):
-    """
-    Checks if all 3 positions in given combination are occupied by given player.
-
-    :param board: Game board.
-    :param combination: Tuple containing three position elements.
-                        Example: ((0,0), (0,1), (0,2))
-
-    Returns True of all three positions in the combination belongs to given
-    player, False otherwise.
-    """
-    pass
+    vals_in_combination = [board[pos[0]][pos[1]] for pos in combination]
+    return len(set(vals_in_combination)) == 1
```

Comment:
Clever use of set to check that there is one player occupying all the spots in a winning combination!